hist/hist/src/TH1.cxx

```diff
@@ -9928,6 +9928,23 @@ TH1F::~TH1F()
 {
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Increment bin content by 1.
+
+void TH1F::AddBinContent(Int_t bin)
+{
+   if (fArray[bin] < 1e7) ++fArray[bin];
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Increment bin content by w.
+
+void TH1F::AddBinContent(Int_t bin, Double_t w)
+{
+   const Float_t newval = fArray[bin] + Float_t(w);
```

Comment:
Conversion from `double` to `float` should be guarded against overflows probably, since this PR is dealing with that same topic

Reply:
This is in analogy to https://root.cern.ch/doc/master/TH1_8cxx_source.html#l09559
There is no extra check there.

Usually, the user is expected to use a weight for addbincontent that is compatible with the underlying TH1.

Of course, it would be better to have a templated class where the weight matches with the TH1<type> class.

Reply:
Makes sense, we can leave it as it is :+1: